pwa/app/routes/district.$abbreviation.($year).tsx

```diff
@@ -0,0 +1,555 @@
+import { LoaderFunctionArgs } from '@remix-run/node';
+import {
+  ClientLoaderFunctionArgs,
+  MetaFunction,
+  Params,
+  useLoaderData,
+} from '@remix-run/react';
+import { countBy, groupBy, sumBy } from 'lodash-es';
+import { useMemo } from 'react';
+
+import {
+  Award,
+  DistrictRanking,
+  Event,
+  LeaderboardInsight,
+  Match,
+  Team,
+  getDistrictAwards,
+  getDistrictEvents,
+  getDistrictHistory,
+  getDistrictRankings,
+  getDistrictTeams,
+  getEventMatches,
+} from '~/api/v3';
+import { TitledCard } from '~/components/tba/cards';
+import { DataTable } from '~/components/tba/dataTable';
+import { Leaderboard } from '~/components/tba/leaderboard';
+import { EventLink, LocationLink, TeamLink } from '~/components/tba/links';
+import { Badge } from '~/components/ui/badge';
+import { Divider } from '~/components/ui/divider';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableFooter,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from '~/components/ui/table';
+import { Tabs, TabsContent, TabsList, TabsTrigger } from '~/components/ui/tabs';
+import { AwardType } from '~/lib/api/AwardType';
+import { EventType } from '~/lib/api/EventType';
+import { getEventDateString, sortEvents } from '~/lib/eventUtils';
+import { sortTeams } from '~/lib/teamUtils';
+import {
+  USA_STATE_ABBREVIATION_TO_FULL,
+  joinComponents,
+  parseParamsForYearElseDefault,
+} from '~/lib/utils';
+
+async function loadData(params: Params) {
+  if (params.abbreviation === undefined) {
+    throw new Error('missing abbreviation');
+  }
+
+  const year = await parseParamsForYearElseDefault(params);
+  if (year === undefined) {
+    throw new Error('invalid year');
+  }
+
+  const [districtHistory, rankings, teams, events, awards] = await Promise.all([
+    await getDistrictHistory({ districtAbbreviation: params.abbreviation }),
+    await getDistrictRankings({ districtKey: `${year}${params.abbreviation}` }),
+    await getDistrictTeams({ districtKey: `${year}${params.abbreviation}` }),
+    await getDistrictEvents({ districtKey: `${year}${params.abbreviation}` }),
+    await getDistrictAwards({ districtKey: `${year}${params.abbreviation}` }),
+  ]);
+
+  if (
+    districtHistory.status !== 200 ||
+    rankings.status !== 200 ||
+    teams.status !== 200 ||
+    events.status !== 200 ||
+    awards.status !== 200
+  ) {
+    throw new Response(null, { status: 404 });
+  }
+
+  // The api returns a lot of teams that previously played in the district but didn't in the given year
+  const actuallyActiveRankings =
```

Comment:
Can/should this filtering be done not in the fetching code, but in the render?

Reply:
IMO it should stay in the fetch because it's a workaround for a backend bug. The API shouldn't be returning the data that it does, and until we fix it, I think we can mediate it here.